flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -297,6 +297,11 @@ public void checkpointState(
         try {
             if (takeSnapshotSync(
                     snapshotFutures, metadata, metrics, options, operatorChain, isCanceled)) {
+                LOG.debug(
+                        "Is unaligned checkpoint for task {} : {}",
```

Comment:
I think a better place to log is the end of `takeSnapshotSync` where other properties are logged.

Reply:
I moved both the log and metrics update to the end of `takeSnapshotSync `.